pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -489,6 +494,18 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 		return fmt.Errorf("Loadbalancer %s does not exist", loadBalancerName)
 	}
 
+	// Get all listeners for this loadbalancer
+	err = listeners.List(lbaas.network, listeners.ListOpts{LoadbalancerID: loadbalancer.ID}).EachPage(func(page pagination.Page) (bool, error) {
+		listenersList, err = listeners.ExtractListeners(page)
```

Comment:
`listenersList` is replaced here, so will only contain the listeners from the last page after this loop!